File: BehaviorPattern/Mediator/Python/concretes/chat_mediator_impl.py
from interfaces import ChatMediator

class ChatMediatorImpl(ChatMediator):
    def __init__(self):
        self.users = []

    def send_message(self, message, user):
        for u in self.users:
            if u != user:
                u.receive(message)

    def add_user(self, user):
        self.users.append(user)

File: BehaviorPattern/Mediator/Golang/concretes/chat_mediator_impl.go
package concretes

import "mediator_pattern/interfaces"

// ChatMediatorImpl is a concrete implementation of the ChatMediator interface.
type ChatMediatorImpl struct {
	users []interfaces.User
}

// NewChatMediatorImpl creates a new ChatMediatorImpl.
func NewChatMediatorImpl() *ChatMediatorImpl {
	return &ChatMediatorImpl{}
}

// SendMessage sends a message to all users except the sender.
func (c *ChatMediatorImpl) SendMessage(message string, user interfaces.User) {
	for _, u := range c.users {
		if u != user {
			u.Receive(message)
		}
	}
}

// AddUser adds a user to the mediator.
func (c *ChatMediatorImpl) AddUser(user interfaces.User) {
	c.users = append(c.users, user)
}
